pkg/storageops/vsphere/vsphere.go

```diff
@@ -91,6 +95,59 @@ func (ops *vsphereOps) Create(opts interface{}, labels map[string]string) (inter
 		return nil, err
 	}
 
+	isPod, storagePod, err := IsStoragePod(ctx, vmObj, volumeOptions.Datastore)
+	if err != nil {
+		return nil, err
+	}
+
+	if isPod {
+		logrus.Infof("Using storage pod: %s", storagePod.Name())
+		var devices object.VirtualDeviceList
+		scsi, err := devices.CreateSCSIController("scsi")
+		if err != nil {
+			return nil, err
+		}
+
+		devices = append(devices, scsi)
+
+		controller, err := devices.FindDiskController("scsi")
+		if err != nil {
+			return nil, err
+		}
+
+		disk := &types.VirtualDisk{
+			VirtualDevice: types.VirtualDevice{
+				Key: devices.NewKey(),
+				Backing: &types.VirtualDiskFlatVer2BackingInfo{
+					DiskMode:        string(types.VirtualDiskModePersistent),
+					ThinProvisioned: types.NewBool(true),
+				},
+			},
+			CapacityInKB: int64(volumeOptions.CapacityKB),
+		}
+
+		devices = append(devices, disk)
```

Comment:
nit: Can you add a doc as to why we are adding the ScsciController and the VirtualDisk to the same VirtualDeviceList?

Reply:
Added below comment in `getDatastoreToUseInStoragePod()`

```
   // devices is a list of devices in the virtual machine (disks and disk controllers) that
   // will be part of the request spec to storage resource manager
   var devices object.VirtualDeviceList


```